pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -469,37 +468,42 @@ func anonymizeLog(stdout io.Writer, logFile string, anonymizer *anonymizer) erro
 // to anonymize.  It currently works for pods owned by StatefulSets,
 // ReplicaSets, and Deployments.
 func getSignificantPodNames(podName string) []string {
-	subNames := make([]string, 2)
-	nameComponents := strings.Split(podName, "-")
-	significantComponent := false
-	for i, component := range nameComponents {
-		// Indicates we found a Pod that comes from either a
-		// ReplicaSet and/or a Deployment
-		if _, err := hex.DecodeString(component); err == nil {
-			significantComponent = true
-		}
-		// Indicates we found a Pod that comes from a StatefulSet
-		if _, err := strconv.Atoi(component); err == nil {
-			significantComponent = true
+	// if the pods ends in an ordinal we can be pretty sure it's
+	// coming from a StatefulSet.
+	statefulSetRegex := regexp.MustCompile("(?P<appName>.*)-([0-9]+)$")
+	// ReplicasSets, and therefore Deployments because they create
+	// ReplicaSets, have a hash followed by a 5 character identity
+	// string attached to the end.
+	replicaSetRegex := regexp.MustCompile("(?P<appName>.*)-(?P<hash>[0-9a-f]+)-([0-9a-z]{5})$")
+	sigNames := []string{}
+	switch {
+	case statefulSetRegex.MatchString(podName):
```

Comment:
this seems ridiculously complicated to get matches but I couldn't find another way to do it?? If another way exists, I'd love to know.